spark/src/main/scala/geotrellis/spark/stitch/StitchCollectionMethods.scala

```diff
@@ -25,17 +27,27 @@ import geotrellis.util._
 
 import org.apache.spark.rdd.RDD
 
-abstract class SpatialTileLayoutCollectionStitchMethods[V <: CellGrid: Stitcher, M: GetComponent[?, LayoutDefinition]]
-  extends MethodExtensions[Seq[(SpatialKey, V)] with Metadata[M]] {
+abstract class SpatialTileLayoutCollectionStitchMethods[
+  V <: CellGrid: ? => TilePrototypeMethods[V]: ? => TileMergeMethods[V],
+  M: GetComponent[?, LayoutDefinition]
+] extends MethodExtensions[Seq[(SpatialKey, V)] with Metadata[M]] {
 
   def stitch(): Raster[V] = {
-    val (tile, bounds) = TileLayoutStitcher.stitch(self)
-    val mapTransform = self.metadata.getComponent[LayoutDefinition].mapTransform
-    Raster(tile, mapTransform(bounds))
+    val (tile, (kx, ky), (offsx, offsy)) = TileLayoutStitcher.stitch(self)
+    val layout = self.metadata.getComponent[LayoutDefinition]
+    val mapTransform = layout.mapTransform
+    val nwTileEx = mapTransform(kx, ky)
+    val base = nwTileEx.southEast
+    val (ulx, uly) = (base.x - offsx.toDouble * layout.cellwidth, base.y + offsy * layout.cellheight)
+    Raster(tile, Extent(ulx, uly - tile.rows * layout.cellheight, ulx + tile.cols * layout.cellwidth, uly))
+
+    // val (tile, bounds) = TileLayoutStitcher.stitch(self)
```

Comment:
Do we need this code, and do we need `Stitcher` type class at the moment?

Reply:
`Stitcher` is fairly well embedded in the code base and has a bit different of a usage pattern (specifying pixel locations of tiles) and so is more general.  In fact, the argument could be made that I should have used that object's methods to implement this PR (which wouldn't change the signature).  I can still do that, and probably should, actually.